src/context/AuthContext.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  AuthError,
+  AuthResponse,
+  Session,
+  User,
+  UserAttributes,
+  UserResponse,
+} from '@supabase/supabase-js';
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import supabaseAdmin from '../supabase/createAdminClient';
+import supabase from '../supabase/createClient';
+
+export interface AuthState {
```

Comment:
Can we add a few comments/docstrings so other devs know how this works. Our current flow requires that we simply import queries from the `/queries` folder, so we should make this change clear. They don't have to be long. Perhaps a few sentences explaining what the `AuthState` interface represents, for example.